sssom/util.py

```diff
@@ -712,9 +706,14 @@ def inject_metadata_into_df(msdf: MappingSetDataFrame) -> MappingSetDataFrame:
 
     :return: MappingSetDataFrame with metadata as columns
     """
+    # TODO Check if 'k' is a valid 'slot' for 'mapping' [sssom.yaml]
+    license = "license"
+    mapping_set_id = "mapping_set_id"
     if msdf.metadata is not None and msdf.df is not None:
         for k, v in msdf.metadata.items():
-            if k not in msdf.df.columns:
+            if k not in msdf.df.columns and k != license:
+                if k == mapping_set_id:
+                    k = "mapping_set_source"
```

Comment:
replace with slot (not string!) when https://github.com/mapping-commons/sssom/pull/135